server/middlewares/validateAccess.js

```diff
@@ -0,0 +1,38 @@
+import getRoleId from '../helpers/getRoleId';
+import models from '../models';
+
+const { User } = models;
+/**
+ * This middleware takes care of permissions across the app
+ * @param {array} arrayOfPermissions - an arrayOfPermissions
+ * @returns {void}
+ */
+const validateAccess = arrayOfPermissions => (req, res, next) => {
+  const { id } = req.userData;
+  let roleId, userData;
+  (async () => {
+    try {
+      roleId = await getRoleId(id);
+      userData = await User.findByPk(id);
+      req.userData = userData;
```

Comment:
I just discovered last night that this will not work the way we intended it to. I think this is due to the asynchronous nature of promises. By the time `req.userData` object here gets populated, the controller/middleware after this `validateAccess` middleware will have already started running. It doesn't stop for this `await` statement on line 16 to resolve before continuing on to the next middleware.

This simply means `req.userData` will not have the values we are adding here in the next middleware because the promise hasn't resolved yet. We need to discuss and find a better solution.